### PR TITLE
Cleanup and remove warnings

### PR DESCRIPTION
**Pull request by TomFischer**
While searching a bug I raised the warning level of the compiler. PR tries to fix some warnings.

### CODE REVIEW

**Comment by endJunction** [comment]
:fast_forward:

File: GeoLib/Polyline.cpp [hunk 1]
```diff
@@ -94,7 +94,9 @@ void Polyline::insertPoint(std::size_t pos, std::size_t pnt_id)
 		}
 	}
 
-	std::vector<std::size_t>::iterator it(_ply_pnt_ids.begin() + pos);
+	std::vector<std::size_t>::iterator it(
+	    _ply_pnt_ids.begin() +
+	    static_cast<std::vector<std::size_t>::difference_type>(pos));
```
Comment (chleh) [inline GeoLib/Polyline.cpp]: Minor thing: I think the code would be much more readable if you you'd provide a `const pos_signed` once in this method and would not cast many times. Or does `pos` change? At least it's not const.

**Comment by chleh** [comment]
:+1:

**Comment by chleh** [comment]
A side point: Maybe we should remove `-Wno-deprecated` from `CompilerSetup.cmake`
```
set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} ${CPU_FLAGS} -std=c++11 -Wno-deprecated -Wall -Wextra")
```
Or will that blow up the warning list?

**Comment by endJunction** [comment]
I've added some more of the explicit ctor/copy/assignment as an addendum here.

@chleh Eigen has some of, now deprecated, `throw()` things attached to some functions; noexcept would be c++11 standard. There are less then 100 warnings or so.

**Comment by endJunction** [comment]
Jenkins, test this please!

Some fetching errors Jenkins' side....

**Comment by endJunction** [comment]
Jenkins, test this please!

Some fetching errors Jenkins' side....

**Pull request merged**